src/Http/Controllers/OAuthController.php

```diff
@@ -43,6 +43,9 @@ protected function successRedirectUrl()
 
         parse_str($query, $query);
 
-        return array_get($query, 'redirect', $default);
+        $referer = array_get($query, 'redirect', $default);
+
+        // Only use the referer, if it's a url to the control panel.
+        return Str::contains($referer, '/'.config('statamic.cp.route')) ? $referer : $default;
```

Comment:
In what situation could you have a non CP URL?

Reply:
I wondered as well, but this happens more often than i thought myself (tested it). 

Happens for the normal login redirect as well. 

This prevents it from it, so I used the same logic to align those behaviours: 
https://github.com/statamic/cms/blob/3.4/src/Http/Controllers/CP/Auth/LoginController.php#L102-L107

Reply:
I see. Thanks.